src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.lib.exception.DataProcessingException;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    private static final SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            assert transaction != null;
```

Comment:
let's stick to simple `if`. if this assertion is not true - AssertionError will be thrown and the next exception will be unavailable, so we won't even know what happened